### DISCUSSION TITLE
UC methods like sb.uc_gui_handle_captcha() and sb.uc_gui_click_captcha() dont working on website

### DISCUSSION THREAD

**Discussion by armorsmith-tech** [seleniumbase/SeleniumBase#3157] [General]
I wrote the simple code:

```
import sys
import time
from seleniumbase import SB


# An incomplete UserAgent forces CAPTCHA-solving on macOS
agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) Chrome/126.0.0.0"
if "linux" in sys.platform or "win32" in sys.platform:
    agent = None  # Use the default UserAgent


with SB(uc=True, test=True, incognito=True, locale_code="en") as sb:
    url = "https://bartio.faucet.berachain.com/"
    sb.uc_open_with_reconnect(url, 4)
    sb.find_element(selector='terms', by='id').uc_click()
    sb.uc_click('button:contains("I AGREE")')

    sb.uc_gui_handle_captcha()
    sb.assert_element("img#captcha-success", timeout=3)

    sb.set_messenger_theme(location="bottom_center")
    sb.post_message("SeleniumBase wasn't detected!")
```

In all of the examples in the documentation, everything worked for me.
but with this site I had a problem
driver just dont put the flag on, IDK why

![image](https://github.com/user-attachments/assets/395a2832-43ce-41cf-909a-d0bef84e5988)

I have this alert on the site, I skipping it in my code

I also tried to found like the iframe of cloudflare but it didnt help me
any ideas?

**Comment by mdmintz** [comment]
This applies to any site:
You may need to use `uc_gui_click_captcha()` instead of `uc_gui_handle_captcha()`.
Run with `--debug` to get the coordinates of the click so that you can call `uc_gui_click_x_y(x, y)`.
There's a new page refresh that happens before the click because Selenium gets detected now when finding the location of the CAPTCHA checkbox. You can avoid the page refresh by calling `uc_gui_click_x_y(x, y)` from a disconnected state. (See the UC Mode methods for keeping the driver disconnected from Chrome.)

**Reply by armorsmith-tech** [reply]
thanks, as I tried it is working good

**Reply by armorsmith-tech** [reply]
I've run into a new problem
how do i do the same thing in --headless mode? will displays/anything else help me?

**Reply by mdmintz** [reply]
`PyAutoGUI` doesn't work in headless mode. A headed browser must be used.

**Reply by armorsmith-tech** [reply]
Nah. Any captcha bypass based on this? I'll have to buy another server with browser display support :(

**Reply by mdmintz** [reply]
It works on Linux with a regular browser because of the special virtual display. `xvfb=True`
